src/signer/default-signer.ts

```diff
@@ -0,0 +1,147 @@
+    this.provider = provider;
+  }
+
+  /**
+   * Returns the checksum address to sign messages with.
+   * If the address is not given, the default address of the wallet is used.
+   * @param {string} address - The address of the account to sign the message with.
+   */
+  getAddress(address?: string): string {
+    return this.wallet.getImpliedAddress(address);
+  }
+
+  /**
+   * Signs a message with the private key of the given address.
+   * If an address is not given, the default address of the wallet is used.
+   * @param {string} message - The message to sign.
+   * @param {string} address - The address of the account to sign the message with.
+   */
+  signMessage(message: string, address?: string): Promise<string> | string {
+    return this.wallet.sign(message, address);
+  }
+
+  /**
+   * Signs and sends a transaction to the network
+   * @param {TransactionInput} transactionObject
+   * @param {boolean} isDryrun - dryrun option.
+   * @return {Promise<any>}
+   */
+  async sendTransaction(transactionObject: TransactionInput, isDryrun: boolean = false) {
+    const txBody = await this.buildTransactionBody(transactionObject);
+    const signature = await this.wallet.signTransaction(txBody, transactionObject.address);
+    return await this.sendSignedTransaction(signature, txBody, isDryrun);
+  }
+
+  /**
+   * Sends signed transactions to the network.
+   * @param {TransactionInput[]} transactionObjects
+   */
+  async sendTransactionBatch(transactionObjects: TransactionInput[]): Promise<any> {
+    let promises: Promise<any>[] = [];
+    for (let tx of transactionObjects) {
+      promises.push(this.buildTransactionBody(tx).then(async (txBody) => {
+        if (tx.nonce === undefined) {
+          // Batch transactions' nonces should be specified.
+          // If they're not, they default to un-nonced (nonce = -1).
+          txBody.nonce = -1;
+        }
+        const signature = await this.wallet.signTransaction(txBody, tx.address);
+        return { signature, tx_body: txBody };
+      }));
+    }
+    return Promise.all(promises).then(async (tx_list) => {
+      const resultList = await this.provider.send('ain_sendSignedTransactionBatch', { tx_list });
+      if (!Array.isArray(resultList)) {
+        return resultList;
+      }
+      const len = resultList.length;
+      if (len !== tx_list.length) {
+        return resultList;
+      } else {
+        for (let i = 0; i < len; i++) {
+          if (!resultList[i] || typeof resultList[i] !== 'object') {
+            resultList[i] = { result: resultList[i] };
+          }
+        }
+        return resultList;
+      }
+    })
+  }
+
+  /**
+   * Sends a signed transaction to the network
+   * @param {string} signature
+   * @param {TransactionBody} txBody
+   * @param {boolean} isDryrun - dryrun option.
+   * @return {Promise<any>}
+   */
+  async sendSignedTransaction(signature: string, txBody: TransactionBody, isDryrun: boolean = false): Promise<any> {
```

Comment:
I moved this function to reduce code duplicated. Please comment about this. (But, this function does not require sign)